app/controllers/SubmitArrangementController.scala

```diff
@@ -48,16 +52,26 @@ class SubmitArrangementController @Inject() (
     }
   }
 
-  private def submitArrangementAndUpdateJourney(journey: Journey.Stages.AgreedTermsAndConditions)(implicit request: Request[_]): Future[Result] = {
-    for {
-      arrangementResponse <- ttpService
-        .submitArrangement(journey)
-      _ <- journeyService
-        .updateArrangementResponse(journey.id, arrangementResponse)
-        .recover {
-          case e: Exception => Errors.throwServerErrorException(e.getMessage)
-        }
-    } yield Redirect(routes.PaymentPlanSetUpController.paymentPlanSetUp())
+  private def submitArrangementAndUpdateJourney(
+      journey: Journey.Stages.AgreedTermsAndConditions
+  )(implicit request: AuthenticatedJourneyRequest[_]): Future[Result] = {
+    ttpService.submitArrangement(journey)
+      .map { (r: ArrangementResponse) =>
+        auditService.auditPaymentPlanSetUp(journey, Right(r))
```

Comment:
could we perhaps move the audit call to the service layer? It would be closer to the actual event there, plus if we ever need to call the service endpoint from another place then the auditing will already be taken care of. Not that that is very likely to happen though...

Reply:
sure